### PR TITLE
Update kubeadm-upgrade.md

### PR DESCRIPTION
**Pull request by ousiax**
Clean the old YUM `kubernetes` repo cache.

### CODE REVIEW

**Comment by AmarNathChary** [comment]
**_Deployment preview_** : https://deploy-preview-45271--kubernetes-io-main-staging.netlify.app/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/#determine-which-version-to-upgrade-to

**Comment by stmcginnis** [comment]
Can you add some explanation as to why you would want to do this? I don't think most users will want or benefit from doing a clean.

**Review by neolit123 — COMMENTED** [review]
/sig cluster-lifecycle

**Comment by ousiax** [comment]
> Can you add some explanation as to why you would want to do this? I don't think most users will want or benefit from doing a clean.

It's a litte weird. But when the minor of the `kubernetes` repo was updated, e.g. from v1.26 to v1.27 as below, I found the cache of the `kubernetes` repo should be cleaned like `apt-get update` in Debian-like OS.

```console
$ cat /etc/yum.repos.d/kubernetes.repo 
[kubernetes]
name=Kubernetes
baseurl=https://pkgs.k8s.io/core:/stable:/v1.26/rpm/
enabled=1
gpgcheck=1
gpgkey=https://pkgs.k8s.io/core:/stable:/v1.26/rpm/repodata/repomd.xml.key
exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni 
$ sudo yum clean all --disablerepo="*" --enablerepo=kubernetes
Loaded plugins: fastestmirror
Cleaning repos: kubernetes
Other repos take up 207 M of disk space (use --verbose for details)
$ sudo yum list --showduplicates kubeadm --disableexcludes=kubernetes
Loaded plugins: fastestmirror
Determining fastest mirrors
 * base: mirrors.cqu.edu.cn
 * extras: mirrors.ustc.edu.cn
 * updates: mirrors.cqu.edu.cn
kubernetes                                                                                                                                                | 1.7 kB  00:00:00     
kubernetes/primary                                                                                                                                        |  25 kB  00:00:05     
kubernetes                                                                                                                                                               244/244
Installed Packages
kubeadm.x86_64                                                                   1.26.14-150500.1.1                                                                   @kubernetes
Available Packages
kubeadm.x86_64                                                                   1.26.0-150500.2.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.1-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.2-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.3-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.4-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.5-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.6-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.7-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.8-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.9-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.10-150500.1.1                                                                   kubernetes 
kubeadm.x86_64                                                                   1.26.11-150500.1.1                                                                   kubernetes 
kubeadm.x86_64                                                                   1.26.12-150500.1.1                                                                   kubernetes 
kubeadm.x86_64                                                                   1.26.13-150500.1.1                                                                   kubernetes 
kubeadm.x86_64                                                                   1.26.14-150500.1.1                                                                   kubernetes 
$ sudo sed -i 's/v1.26/v1.27/' /etc/yum.repos.d/kubernetes.repo 
$ cat /etc/yum.repos.d/kubernetes.repo 
[kubernetes]
name=Kubernetes
baseurl=https://pkgs.k8s.io/core:/stable:/v1.27/rpm/
enabled=1
gpgcheck=1
gpgkey=https://pkgs.k8s.io/core:/stable:/v1.27/rpm/repodata/repomd.xml.key
exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni 
$ sudo yum list --showduplicates kubeadm --disableexcludes=kubernetes
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirrors.cqu.edu.cn
 * extras: mirrors.ustc.edu.cn
 * updates: mirrors.cqu.edu.cn
Installed Packages
kubeadm.x86_64                                                                   1.26.14-150500.1.1                                                                   @kubernetes
Available Packages
kubeadm.x86_64                                                                   1.26.0-150500.2.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.1-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.2-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.3-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.4-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.5-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.6-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.7-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.8-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.9-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.26.10-150500.1.1                                                                   kubernetes 
kubeadm.x86_64                                                                   1.26.11-150500.1.1                                                                   kubernetes 
kubeadm.x86_64                                                                   1.26.12-150500.1.1                                                                   kubernetes 
kubeadm.x86_64                                                                   1.26.13-150500.1.1                                                                   kubernetes 
kubeadm.x86_64                                                                   1.26.14-150500.1.1                                                                   kubernetes 
$ sudo yum clean all --disablerepo="*" --enablerepo=kubernetes
Loaded plugins: fastestmirror
Cleaning repos: kubernetes
Cleaning up list of fastest mirrors
Other repos take up 207 M of disk space (use --verbose for details)
$ sudo yum list --showduplicates kubeadm --disableexcludes=kubernetes
Loaded plugins: fastestmirror
Determining fastest mirrors
 * base: mirrors.163.com
 * extras: mirrors.bfsu.edu.cn
 * updates: mirrors.bfsu.edu.cn
kubernetes                                                                                                                                                | 1.7 kB  00:00:00     
kubernetes/primary                                                                                                                                        |  21 kB  00:00:03     
kubernetes                                                                                                                                                               204/204
Installed Packages
kubeadm.x86_64                                                                   1.26.14-150500.1.1                                                                   @kubernetes
Available Packages
kubeadm.x86_64                                                                   1.27.0-150500.2.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.27.1-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.27.2-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.27.3-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.27.4-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.27.5-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.27.6-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.27.7-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.27.8-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.27.9-150500.1.1                                                                    kubernetes 
kubeadm.x86_64                                                                   1.27.10-150500.1.1                                                                   kubernetes 
kubeadm.x86_64                                                                   1.27.11-150500.1.1                                                                   kubernetes         
```

**Comment by neolit123** [comment]
/sig release k8s-infra
i'm not familiar with how the cache is supposed to behave.

**Comment by ousiax** [comment]
It seems that this PR is pending now. What could I do to make any progress?

**Comment by stmcginnis** [comment]
OK, I see `--disablerepo="*" --enablerepo=kubernetes` will make sure that only the kubernetes repo is cleaned. So that's probably a pretty harmless thing to do, that gets around the issue you point out.

It might be good to add a comment above that line with a brief note stating what/why someone would want to run that line, but I don't think it's critical.

/lgtm

**Comment by niranjandarshann** [comment]
@ousiax  According to me this  command is used to clean the  cache while disabling all repositories except for the "kubernetes" repository, which remains enabled. This might be useful in scenarios where you want  installing or updating packages related to Kubernetes.
According to me it will be good as it free us Disk Space as well.
Looks Good to me 
/lgtm

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade.md [hunk 1]
```diff
@@ -85,6 +85,7 @@ sudo apt-cache madison kubeadm
 ```shell
 # Find the latest {{< skew currentVersion >}} version in the list.
 # It should look like {{< skew currentVersion >}}.x-*, where x is the latest patch.
+sudo yum clean all --disablerepo="*" --enablerepo=kubernetes # Make sure the YUM cache of the kubernetes repo is cleaned.
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade.md:88]: I'd like SIG Cluster Lifecycle to confirm that this is what we expect to document.

A blanket recommendation may have cost implications for SIG K8s Infra, too (people pop this sort of thing in cron jobs).

**Comment by neolit123** [comment]
this is more of a sig release / k8s infra topic as they maintain the k8s package repos.

/sig release k8s-infra

**Comment by sftim** [comment]
/hold

Waiting on a formal tech review. OK to unhold once we have tech LGTM.

**Comment by divya-mohan0209** [comment]
@kubernetes/sig-release-leads @kubernetes/sig-k8s-infra-leads : Please could you review this PR from a technical perspective?

**Pull request closed without merging**